src/lib/repeat.ts

```diff
@@ -43,112 +49,66 @@ export function repeat<T>(
         const range = document.createRange();
         range.setStartBefore(itemPart.startNode);
-        range.setEndBefore(itemPart.endNode);
-
-        if (currentMarker === undefined) {
-          // this should be the first part, make sure it's first
-          if (part.startNode.nextSibling !== itemPart.startNode) {
-            // move the whole part
-            // get previous and next parts
-            const previousPart = endParts.get(itemPart.startNode);
-            if (previousPart) {
-              previousPart.endNode = itemPart.endNode;
-              endParts.set(previousPart.endNode, previousPart);
-            }
-            const contents = range.extractContents();
-            if (part.startNode.nextSibling === part.endNode) {
-              // The container part was empty, so we need a new endPart
-              itemPart.endNode = document.createTextNode('');
-              container.insertBefore(
-                  itemPart.endNode, part.startNode.nextSibling);
-            } else {
-              // endNode should equal the startNode of the currently first part
-              itemPart.endNode = part.startNode.nextSibling!;
-            }
-            container.insertBefore(contents, part.startNode.nextSibling);
-          }
-          // else part is in the correct position already
-        } else if (currentMarker !== itemPart.startNode) {
-          // move to correct position
-          const previousPart = endParts.get(itemPart.startNode);
-          if (previousPart) {
-            previousPart.endNode = itemPart.endNode;
-            endParts.set(previousPart.endNode, previousPart);
-          }
-          const contents = range.extractContents();
-          container.insertBefore(contents, currentMarker);
-        }
-
-        // remove part from oldParts list so it's not cleaned up
-        oldParts.splice(oldParts.indexOf(itemPart), 1);
+        range.setEndAfter(itemPart.endNode);
+        const contents = range.extractContents();
+        container.insertBefore(contents, currentMarker);
+      } else {
+        // else part is in the correct position already
+        currentMarker = itemPart.endNode.nextSibling;
       }
-      // else part is in the correct position already
 
       itemPart.setValue(result);
       itemParts.push(itemPart);
-      currentMarker = itemPart.endNode;
     }
 
     // Cleanup
-    if (oldParts.length > 0) {
-      const clearStart = oldParts[0].startNode;
-      const clearEnd = oldParts[oldParts.length - 1].endNode;
-      const clearRange = document.createRange();
-      if (itemParts.length === 0) {
-        clearRange.setStartBefore(clearStart);
-      } else {
-        clearRange.setStartAfter(clearStart);
-      }
-      clearRange.setEndAfter(clearEnd);
-      clearRange.deleteContents();
-      clearRange.detach();  // is this neccessary?
+    if (currentMarker !== part.endNode) {
+      const range = document.createRange();
```

Comment:
We realized in lit core that using a Range to clear nodes is really slow. Want to change this to just iterate and remove?

Reply:
https://jsperf.com/range-deletecontents-vs-loop

It seems creating the range is stupid slow, but either way using a loop is faster.

Reply:
Seems extraction is slower, too: https://jsperf.com/range-extractcontents-vs-loop